dataset/src/main/scala/frameless/ops/AggregateTypes.scala

```diff
@@ -0,0 +1,19 @@
+package frameless
+package ops
+
+import shapeless._
+
+trait AggregateTypes[V, U <: HList] {
```

Comment:
It looks like `Comapped` doesn't work for such cases:
```scala
  trait Foo[A, B]

  implicitly[
    Comapped.Aux[
      Foo[String, Int] :: HNil, 
      Lambda[X => Foo[String, X]], 
      Int :: HNil
    ]
  ]
```

I've come up with a dirty workaround:

```scala
  /**
    * Type class witnessing that the result of stripping type constructor `F` off each element of `HList` `L` is `Out`.
    */
  trait Comapped2[A, L <: HList, F[_, _]] extends Serializable {
    type Out <: HList
  }

  trait LowPriorityComapped2 {
    type Id2[A,B] = B
    type Aux[A, L <: HList, F[_, _], Out0 <: HList] = Comapped2[A, L, F] {type Out = Out0}
    implicit def hlistIdComapped[A, L <: HList]: Aux[A, L, Id2, L] = new Comapped2[A, L, Id2] {type Out = L}
  }

  object Comapped2 extends LowPriorityComapped2 {
    def apply[A, L <: HList, F[_, _]](implicit comapped: Comapped2[A, L, F]): Aux[A, L, F, comapped.Out] = comapped

    implicit def hnilComapped[A, F[_, _]]: Aux[A, HNil, F, HNil] = new Comapped2[A, HNil, F] {type Out = HNil}

    implicit def hlistComapped[A, H, T <: HList, F[_, _]](implicit mt: Comapped2[A, T, F]): Aux[A, F[A, H] :: T, F, H :: mt.Out] =
      new Comapped2[A, F[A, H] :: T, F] {type Out = H :: mt.Out}
  }
```

And it works!

```scala
  def agg[TC <: HList, C <: HList, Out0 <: HList, Out1](columns: TC)(
    implicit
    mapped: Comapped2.Aux[T, TC, TypedAggregate, C],
    encoder: TypedEncoder[Out1],
    append: Prepend.Aux[K, C, Out0],
    toTuple: Tupler.Aux[Out0, Out1],
    columnsToList: ToTraversable.Aux[TC, List, UntypedExpression[T]]
  ): TypedDataset[Out1] = {
```